lib/api/core/models/repositories/repository.js

```diff
@@ -217,40 +227,34 @@ Repository.prototype.loadFromCache = function (id, opts) {
  * @returns {Promise}
  */
 Repository.prototype.load = function (id, opts) {
-  var deferred = q.defer();
-
   if (!this.cacheEngine) {
     return this.loadOneFromDatabase(id);
   }
 
-  this.loadFromCache(id, opts)
+  return this.loadFromCache(id, opts)
     .then(object => {
       if (object === null) {
-        if (!this.readEngine) {
-          deferred.resolve(null);
-          return deferred.promise;
+        if (this.readEngine === null) {
+          return null;
         }
-        this.loadOneFromDatabase(id)
+
+        return this.loadOneFromDatabase(id)
           .then(objectFromDatabase => {
             if (objectFromDatabase !== null) {
               this.persistToCache(objectFromDatabase);
             }
-            deferred.resolve(objectFromDatabase);
+            return objectFromDatabase;
           })
           .catch(err => {
-            deferred.reject(err);
+            return q.reject(err);
```

Comment:
nitpicking: this `catch` can be rejected, as it's strictly equivalent to letting the whole chain being rejected.